src/app/portlets/dot-edit-page/content/components/dot-edit-page-toolbar/dot-edit-page-toolbar.component.ts

```diff
@@ -45,23 +45,19 @@ export class DotEditPageToolbarComponent implements OnInit {
                 'editpage.toolbar.page.locked.by.user',
                 'editpage.toolbar.primary.workflow.actions',
                 'dot.common.message.pageurl.copied.clipboard',
-                'dot.common.message.pageurl.copied.clipboard.error'
+                'dot.common.message.pageurl.copied.clipboard.error',
+                'editpage.toolbar.page.cant.edit'
             ])
             .subscribe((res) => {
-                this.states = [
-                    {
-                        label: res['editpage.toolbar.edit.page'],
-                        value: PageMode.EDIT,
-                        styleClass: !this.page.canLock ? 'edit-page-toolbar__state-selector-item--disabled' : ''
-                    },
-                    { label: res['editpage.toolbar.preview.page'], value: PageMode.PREVIEW },
-                    { label: res['editpage.toolbar.live.page'], value: PageMode.LIVE }
-                ];
+                this.workflowsActions = this.getWorkflowOptions();
+                this.setFieldsModels(this.page);
             });
+    }
 
-        this.workflowsActions = this.getWorkflowOptions();
-
-        this.lockerModel = this.page.lockedByAnotherUser && this.page.canLock ? false : this.page.locked;
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes.page && !changes.page.firstChange) {
+            this.setFieldsModels(changes.page.currentValue);
```

Comment:
page can change now... because if any of the set state request (lock/unlock, get preview or edit) fails we need to get page again, this can happen with stuff like: you have permissions to edit a page but someone took away those permissions exactly when you try to edit... is a long shot, but still handle this will handle any requests errors we get.